addons/website_mass_mailing/data/ir_model_data.xml

```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="mass_mailing.model_mailing_contact" model="ir.model">
+            <field name="website_form_key">create_mailing_contact</field>
+            <field name="website_form_access">True</field>
+            <field name="website_form_label">Subscribe to Newsletter</field>
+        </record>
+
+        <function model="ir.model.fields" name="formbuilder_whitelist">
+            <value>mailing.contact</value>
+            <value eval="[
+                'name',
+                'company_name',
+                'title_id',
+                'email',
+                'list_ids',
```

Comment:
Note that we should check what the customer is giving to us (for example he cannot subscribe to non public mailing lists by bidouilling the list_ids)